### PR TITLE
chore: add demo toolings

### PR DESCRIPTION
**Pull request by Foxandxss**
Broke the other PR (by deleting my fork due a problem).

I tried the <script> tag solution. It requires lot of scripting to move the file around. Much better to simply import it and use it.

That is the less ugly solution I could find.

### CODE REVIEW

File: demo/src/vendor.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import 'angular2/bundles/angular2-polyfills';
+
+import 'angular2/platform/browser';
+import 'angular2/core';
+import 'angular2/http';
+import 'angular2/router';
```
Comment (wesleycho) [inline demo/src/vendor.ts]: I don't think the http and router imports are necessary at this stage.

Comment (pkozlowski-opensource) [inline demo/src/vendor.ts]: +1 - let's only import what we use

File: demo/src/vendor.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import 'angular2/bundles/angular2-polyfills';
```
Comment (pkozlowski-opensource) [inline demo/src/vendor.ts]: I thought that we had a discussion about it :-) I need to fix it in all the other seed repos.

File: webpack.demo.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+var webpack = require('webpack');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  entry: {
+    'app': './demo/src/demo.ts',
+    'vendor': './demo/src/vendor.ts'
```
Comment (pkozlowski-opensource) [inline webpack.demo.js]: I really wouldn't bother about splitting it into vendor right now.

File: webpack.demo.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        test: /\.ts$/,
+        loader: 'ts-loader'
+      }
+    ],
+    noParse: [ /angular2\/bundles\/.+/ ]
+  },
+  plugins: [
+    new webpack.optimize.OccurrenceOrderPlugin(),
+    new webpack.optimize.DedupePlugin(),
+    new webpack.optimize.CommonsChunkPlugin({
```
Comment (pkozlowski-opensource) [inline webpack.demo.js]: So as soon as you use commons chunk plugin all the benefits of excluding vendor go out of the window. My vote: let's not bother and keep webpack's config simple

Comment (pkozlowski-opensource) [inline webpack.demo.js]: We can always make it more complex if / when needed.

File: demo/src/vendor.ts (hunk 1 above)
Comment (Foxandxss) [inline demo/src/vendor.ts]: The thing is that angular imports them anyway and it ends on the app bundle. By importing them by hand, they end on the vendor bundle.

TL;DR I can remove those two but will be on the bundle anyway. I will need that for the demo tho.

File: demo/src/vendor.ts (hunk 2 above)
Comment (Foxandxss) [inline demo/src/vendor.ts]: <script> tag is -1 for me.

File: webpack.demo.js (hunk 3 above)
Comment (Foxandxss) [inline webpack.demo.js]: I don't mind that. We can require `vendor.ts` on `demo.ts`.

**Comment by Foxandxss** [comment]
I deleted the extra vendor bundle. Now you have to require `vendor.ts` on the `demo.ts`.

File: demo/src/vendor.ts (hunk 2 above)
Comment (pkozlowski-opensource) [inline demo/src/vendor.ts]: Soooo, we _really_ don't want to do this, see https://github.com/angular/angular/issues/5980

Comment (Foxandxss) [inline demo/src/vendor.ts]: It is not failing for me at all.

Comment (pkozlowski-opensource) [inline demo/src/vendor.ts]: It just means that you don't see issue with your current use case. But there are problems. If I can't convince you maybe Misko will? https://github.com/angular/angular/issues/5304#issuecomment-158133227

Comment (Foxandxss) [inline demo/src/vendor.ts]: It is not about trust. It was always working and everybody was doing it too without problems.

I will change it and write some scripts for deployment.

Comment (pkozlowski-opensource) [inline demo/src/vendor.ts]: >  It was always working and everybody was doing it too without problems.

It is more that you didn't have use-cases that would expose problems. This doesn't mean that those don't exist.

Comment (Foxandxss) [inline demo/src/vendor.ts]: agree on that

**Comment by Foxandxss** [comment]
Review please.

File: demo/index.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<!doctype html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>ng-bootstrap</title>
+</head>
+<body>
+  <demo></demo>
```
Comment (pkozlowski-opensource) [inline demo/index.html]: use `-` in component selectors, ex.: `demo-app`

File: .gitignore [hunk 6]
```diff
@@ -2,6 +2,8 @@
 /typings/
 /dist/
 /temp/
+/demo/dist/
```
Comment (pkozlowski-opensource) [inline .gitignore]: maybe let's use `dist/demo`?

File: webpack.demo.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+var webpack = require('webpack');
+var HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  entry: {
+    'app': [
+      'angular2/core',
```
Comment (pkozlowski-opensource) [inline webpack.demo.js]: why do we need `angular2/core` as an entry here?

File: demo/src/demo.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import {Component} from 'angular2/core';
+import {bootstrap} from 'angular2/bootstrap';
```
Comment (pkozlowski-opensource) [inline demo/src/demo.ts]: import from `angular2/platform/browser` - `angular2/bootstrap` is going away

File: demo/index.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  <script src="lib/angular2-polyfills.js"></script>
```
Comment (pkozlowski-opensource) [inline demo/index.html]: do we want to add `es-shim` as well?

File: .gitignore (hunk 6 above)
Comment (Foxandxss) [inline .gitignore]: Dist folder is something we have to talk about yet. We are not really sure in how are we going to deploy it.

On the other hand, the demo deploy is just a temporary folder. I could even create a task to clean it after success push.

So the idea is, when you need to update the gh-pages, you run a command, `/demo/dist` gets generated and then pushed to gh-pages. After that you don't need and if we decide to put root's `/dist` on NPM, I wouldn't certainly want the demo dist in there.

So it is a -1 for me.

File: webpack.demo.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      './demo/src/demo.ts'
+    ]
+  },
+  output: {
+    path: __dirname + '/demo/dist',
+    filename: '[name].bundle.js',
+    chunkFilename: '[name].bundle.js'
```
Comment (pkozlowski-opensource) [inline webpack.demo.js]: Do we still need `chunkFilename`?

File: webpack.demo.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  devtool: 'eval',
+  module: {
+    loaders: [
+      {
+        test: /\.ts$/,
+        loader: 'ts-loader'
+      }
+    ],
+    noParse: [ /angular2\/bundles\/.+/ ]
```
Comment (pkozlowski-opensource) [inline webpack.demo.js]: I don't think we need this one any more

**Comment by pkozlowski-opensource** [comment]
Left some comments. Good stuff!

File: demo/index.html (hunk 5 above)
Comment (Foxandxss) [inline demo/index.html]: sure

File: webpack.demo.js (hunk 7 above)
Comment (Foxandxss) [inline webpack.demo.js]: seems like we don't.

File: demo/src/demo.ts (hunk 8 above)
Comment (Foxandxss) [inline demo/src/demo.ts]: Good to know.

File: demo/index.html (hunk 9 above)
Comment (Foxandxss) [inline demo/index.html]: I don't have an answer for that tbh.

Comment (Foxandxss) [inline demo/index.html]: Well, we can add it when it gets needed, not for now I think.

File: webpack.demo.js (hunk 10 above)
Comment (Foxandxss) [inline webpack.demo.js]: I can simplify that.

File: webpack.demo.js (hunk 11 above)
Comment (Foxandxss) [inline webpack.demo.js]: Indeed.

**Comment by Foxandxss** [comment]
I addressed all of them (except the dist thing).

**Comment by pkozlowski-opensource** [comment]
LGTM!
